FMWK-598 Implement XDR data parser

filkeith opened the pull request:
- reworked info commands, moved them to existing asinfo package
- improved TCP server, to serve connections more efficiently
- XDR protocol parsed

filkeith committed:
FMWK-598-xdr-data-parser
- server and parser wip

filkeith committed:
FMWK-598-xdr-data-parser
- tcp server

filkeith committed:
FMWK-598-xdr-data-parser
- performance improvments

filkeith committed:
FMWK-598-xdr-data-parser
- connections limiter fix

filkeith committed:
FMWK-598-xdr-data-parser
- ack improvments

filkeith committed:
linter

reugn commented on io/aerospike/xdr/tcp_server.go:
```diff
@@ -17,147 +17,401 @@ package xdr
+		case <-ctx.Done():
+			return
+		default:
+			timeNow := atomic.LoadInt64(&h.timeNow)
+			deadline := timeNow + h.readTimoutNano
```
Those can be extracted to a method and reused.

filkeith committed:
FMWK-598-xdr-data-parser
- move timeout calculation to func

filkeith committed:
FMWK-598-xdr-data-parser
- removed flicking check, as some times server works too fast and already close connection.
